### ISSUE TITLE
cmake build issue

### ISSUE THREAD

**Issue by r-barnes** [arrayfire/arrayfire#2623]
When I build arrayfire like so:
```
CXX=g++-6 CC=gcc-6 cmake -GNinja -DBUILD_TESTING=OFF -DAF_BUILD_EXAMPLES=OFF -DCMAKE_BUILD_TYPE=Release ..
sudo ninja install
```
I later try to use it in a project with CMAKE:

    find_package(ArrayFire REQUIRED)
    ...
    target_link_libraries(mytarget PRIVATE ArrayFire::afcuda)

And disaster strikes:
```
CMake Error at /usr/local/share/ArrayFire/cmake/ArrayFireConfig.cmake:131 (get_property):
  get_property given invalid argument "RELEASE".
Call Stack (most recent call first):
  src/CMakeLists.txt:6 (find_package)
```
So, I rebuild without specifying the build type:
```
CXX=g++-6 CC=gcc-6 cmake -GNinja -DBUILD_TESTING=OFF -DAF_BUILD_EXAMPLES=OFF  ..
sudo ninja install
```
And now things work as expected.

It seems as though using `CMAKE_BUILD_TYPE` creates a problem, tough it is recommended [here](https://github.com/arrayfire/arrayfire/wiki/Build-Instructions-for-Linux).

**Comment by 9prady9** [comment]
What is cmake version you are using ?

**Comment by r-barnes** [comment]
3.10.2, sorry.

**Comment by 9prady9** [comment]
Using cmake 3.10.2, I haven't run into the exact issue you described, but I did face the following problem and I think we aware of this one and this one is related to FindCUDA module which is fixed later in 3.12.2
```
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
CUDA_cublas_device_LIBRARY (ADVANCED)
    linked by target "afcuda" in directory /home/pradeep/gitroot/ArrayFireWorkspace/arrayfire/src/backend/cuda

```
I am going to try on my elementary Juno, will update here once I do.

**Comment by 9prady9** [comment]
I have built CPU backend and installed into a local build folder and later tried to use the installed package to build examples and that worked as expected. 

Can you please do the following - try to create a Dockerfile that can help us replicate your problem in more consistent manner.

**Comment by r-barnes** [comment]
@9prady9 : I'm not sure how to create said Dockerfile for my system.

I did notice that upgrading to cmake 3.15 made `FindCUDA` work better, which was helpful elsewhere.

Since I have the workaround mentioned in the original post, we could close it: I don't really need a fix for this. Mostly I wanted to make sure that the issue+workaround was documented. Happy to try to help figure things out, though.

**Comment by 9prady9** [comment]
I understand your concern, but I wasn't able to reproduce it on either Arch or Ubuntu 18.04, using CMake 3.10.2. I can give it a shot on a fresh docker later. Lets keep this open till then.

**Comment by 9prady9** [comment]
Here is the Dockerfile I tried on fresh bionic image. Once ArrayFire CPU backend is built, I installed into a folder. At this point I interactively logged into the image generated and I built examples using the locally installed ArrayFire. All steps worked as expected.
```Dockerfile
FROM ubuntu:bionic

WORKDIR /

CMD ["/bin/bash"]

RUN apt-get update && \
    apt-get install --yes --no-install-recommends ca-certificates \
            build-essential git cmake libfreeimage-dev \
            libopenblas-dev libfftw3-dev liblapacke-dev \
            opencl-headers libboost-all-dev ocl-icd-opencl-dev

RUN cd / && git clone --recursive https://github.com/arrayfire/arrayfire.git

RUN cmake --version

RUN cd /arrayfire && \
    mkdir build && cd build && \
    cmake .. -DAF_BUILD_OPENCL=OFF -DCMAKE_BUILD_TYPE=Release -DAF_BUILD_EXAMPLES=OFF -DBUILD_TESTING=OFF && \
    make -j4
```

Unfortunately, I haven't been able to reproduce it. Unless you can provide me a Dockerfile with all steps exactly you tried, there isn't much I can debug on my end at this point.

I am closing the issue, please reopen if you are able to create a Dockerfile with steps to reproduce the problem consistently.

Note: The FindCUDA issue has been fixed in 3.12.2 as I have mentioned earlier - just stating in case you someone can't go as high as the latest stable release.
